pkg/instance/instance.go

```diff
@@ -534,15 +545,76 @@ func (i *Instance) RegisterPassphrase(pass, tok []byte) error {
 	if subtle.ConstantTimeCompare(i.RegisterToken, tok) != 1 {
 		return ErrInvalidToken
 	}
-
 	hash, err := crypto.GenerateFromPassphrase(pass)
 	if err != nil {
 		return err
 	}
-
 	i.RegisterToken = nil
 	i.setPassphraseAndSecret(hash)
+	return couchdb.UpdateDoc(couchdb.GlobalDB, i)
+}
 
+// RequestPassphraseReset generates a new registration token for the user to
+// renew its password.
+func (i *Instance) RequestPassphraseReset() error {
+	// If a registration token is set, we do not generate another token than the
+	// registration one, and bail.
+	if i.RegisterToken != nil {
+		return nil
+	}
+	// If a passphrase reset token is set and valid, we do not generate new one,
+	// and bail.
+	if i.PassphraseResetToken != nil &&
```

Comment:
If you have lost your passphrase and ask for a reset, you can only do it once for 24h. So, if you are unlucky and this mail is marked as spam, you have to wait 24h locked out of your cozy. Right? Maybe we can reduce the 24h delay to something a lot shorter, like 15 minutes. I'd also prefer if the passphrase reset token have a short life time for security reasons.